src/ngx_http_lua_uri.c

```diff
@@ -55,8 +60,29 @@ ngx_http_lua_ngx_req_set_uri(lua_State *L)
         return luaL_error(L, "attempt to use zero-length uri");
     }
 
-    if (n == 2) {
+    if (n >= 3) {
+        luaL_checktype(L, 3, LUA_TBOOLEAN);
+        binary = lua_toboolean(L, 3);
+    }
+
+    if (!binary
+        && ngx_http_lua_check_unsafe_uri_bytes(r, p, len, &byte) != NGX_OK)
+    {
+        buf_len = ngx_http_lua_escape_log(NULL, p, len);
+        buf = ngx_palloc(r->pool, buf_len + 1);
```

Comment:
The name `buf_len` is confusing since the real buffer length is actually `buf_len + 1`.